--test--/pet.test.js

```diff
@@ -0,0 +1,174 @@
+const Pet = require("../src/pet");
+
+describe("constructor", () => {
+  it("returns an object", () => {
+    expect(new Pet("Fido")).toBeInstanceOf(Object);
+  });
+
+  it("sets the name property", () => {
+    const pet = new Pet("Fido");
+    expect(pet.name).toEqual("Fido");
+  });
+
+  it("has an initial age of 0", () => {
+    const pet = new Pet("Fido");
+    expect(pet.age).toEqual(0);
+  });
+
+  it("has initial hunger of 0", () => {
+    const pet = new Pet("Fido");
+    expect(pet.hunger).toEqual(0);
+  });
+
+  it("set the children property as an empty array", () => {
+    const pet = new Pet("Fido");
+
+    expect(pet.children).toEqual([]);
+  });
+});
+
+describe("growUp", () => {
+  it("increments the age by 1", () => {
+    const pet = new Pet("Fido");
+    pet.growUp();
+    expect(pet.age).toEqual(1);
+  });
+
+  it("has initial fitness of 10", () => {
+    const pet = new Pet("Fido");
+    expect(pet.fitness).toEqual(10);
+  });
+
+  it("returns true if the pet is alive", () => {
+    const pet = new Pet("Fido");
+    this.age = 29;
+    this.hunger = 9;
+    this.fitness = 1;
+    expect(pet.isAlive).toBe(true);
+  });
+
+  it("returns false if the pet is not alive", () => {
+    const pet = new Pet("Fido");
+    this.age = 30;
+    this.hunger = 10;
+    this.fitness = 0;
+    expect(!pet.isAlive).toBe(false);
+  });
+
+  it("increases hunger property by 5", () => {
+    const pet = new Pet("Fido");
+    pet.growUp();
+    expect(pet.hunger).toEqual(5);
+  });
+  it("decreases the fitness property by 3", () => {
+    const pet = new Pet("Fido");
+    pet.growUp();
+    expect(pet.fitness).toEqual(7);
+  });
+
+  it("throws an error if the pet is not alive", () => {
+    const pet = new Pet("Fido");
+
+    pet.fitness = 0;
+
+    expect(() => pet.growUp()).toThrow("Your pet is no longer alive :(");
```

Comment:
This could all be grouped together:
```
    const pet = new Pet("Fido");
    pet.fitness = 0;
    expect(() => pet.growUp()).toThrow("Your pet is no longer alive :(");
```